### DISCUSSION TITLE
Hello and how can I help?!

### DISCUSSION THREAD

**Discussion by leeprobert** [CariusLars/ar_flutter_plugin#20] [General]
Hey Lars. I'm so stoked you have started putting together this package. I have been struggling to integrate both ARCore and ARKit Flutter packages into my app to the point where I was thinking of trying the Unity ARFoundation hack. I am also currently looking for an Android/ARCore developer to implement some of the features I saw were missing from that package. In particular, I need vertical plane detection and animation support.

Just wondering, do you have a roadmap? Is there a place where you have listed all of the main AR features and the status of support? Might also be worth thinking about Web AR support? There are some interesting opensource developments there too.

Let me know how I can help. I can certainly do some work on the iOS or Web side.

Lee

**Comment by CariusLars** [comment]
Hey Lee,
thanks for your interest! I didn't have a proper roadmap so far, but you're right, the plugin should definitely have one, so I've created a board here: [https://github.com/CariusLars/ar_flutter_plugin/projects/1](https://github.com/CariusLars/ar_flutter_plugin/projects/1). Feel free to propose a different tool if you have a better idea! I don't have a list of all AR features because the possibilities are endless, but the table with example applications in the readme of this repo gives a pretty good overview of what the plugin is currently capable of. 

WebAR would be amazing to have in the plugin if you want to look into that, otherwise I've put a few options for possible contributions on the roadmap board (e.g. image anchors, face anchors, or smaller features like enabling dragging and scaling nodes and adding animations). If you feel like working on one of those, just assign the issue/card to yourself, I'm happy to help you get started or help with the Android counterpart if you work on the iOS side!

By the way: Vertical plane detection should already work with this plugin, just use `PlaneDetectionConfig.horizontalAndVertical` when initializing the `ARSessionManager`.

Cheers

Lars

P.S. if you're looking for a developer in terms of hiring one (not sure if you meant that in the first paragraph of your message), feel free to reach out to me on [LinkedIn](https://www.linkedin.com/in/lars-carius/)

**Reply by leeprobert** [reply]
Fantastic. I'm not sure if you already ported over the `snapshot` method from the `ARKit` package but I had already added that feature in there, so this would be a good place to start. Definitely need that working for both platforms. I'll take a look and fork the repos. 👍

**Reply by CariusLars** [reply]
great, sounds like a good start! I haven't worked on the `snapshot` method so far but I guess it should be fairly easy to build a Flutter wrapper around the ARKit method and then add the ARCore counterpart accordingly, let me know if you need any help :)

**Reply by leeprobert** [reply]
It is easy for ARKit ... but may want to also think about adding video capture as well. Tap and hold to record a video, tap to take a photo. The video capture is not ARKit. Snapshot first then the fancy stuff.
